Satellite remote sensing tuto (#3825)
galaxyproject/training-material: topics/ecology/tutorials/remote-sensing/tutorial.md

>
>    You will have to to create an account for either of these platform. 
>    Select Sentinel 2 and choose the Product type "S2MSI2A".
>
>    ![Scihub portal](../../images/remote_sensing/scihub.png "Scihub portal")
>
>    This an example of the Copernicus portal, Scihub. You need to download a zip folder. Keep it that way. 
>
> 3. Upload the zip folder
>
>     > <tip-title>Tip: Importing data from your computer</tip-title>
>     >
>     > * Open the Galaxy Upload Manager {% icon galaxy-upload %}
>     > * Select **Choose local files**
>     > * Browse in your computer and get the downloaded zip folder
>     >
>     > * Press **Start**
>     {: .tip}
>
> 4. You can rename the dataset, “sentinel_2_data.zip" for example, and to keep informations about the original name "S2A_MSIL2A.....zip", you can copy/paste it on the "Info" box.
>
>
{: .hands_on}

# Preparing data

Using preprocS2 R package, this step provides a unique tool to read, crop, resample the original image directory, and write it as a raster stack

## **Preprocessing sentinel 2 data**

> > <hands-on-title>Hands-on: Preprocess</hands-on-title>

Reviewer: ```suggestion
> <hands-on-title>Preprocess</hands-on-title>
```

The `Hands-on:`, `Question:` etc part are already added automatically, so you can leave them out here and only add what you want to appear after that (or leave empty if nothing)

>
> 1. {% tool [Preprocessing sentinel 2 data](toolshed.g2.bx.psu.edu/repos/ecology/srs_preprocess_s2/srs_preprocess_s2/0.0.1) %} with the following parameters:
>    - {% icon param-file %} *"Input data"*: `sentinel_2_data.zip` (Input dataset)
>    - {% icon param-select %} *"Where does your data come from ?"*: 'From Scihub or Peps'
>
> 2. Click on **Execute** 
>
>    > <comment-title>Comment</comment-title>
>    >
>    > The interesting output is the ENVI image format which is a binary raster file with an accompanying header file. The data are stored as a binary stream of bytes in a BIL file without extension and the metadata are stored in the .hdr file. These data are in the output **Reflectance**. You can directly use the output **Reflectance** for the rest of the analysis.
>    {: .comment}
>
{: .hands_on}

![Preprocessing outputs](../../images/remote_sensing/preproc.png "Preprocessing outputs")

> <question-title>Question</question-title>
>

> > 1. What are the files you are interested in for the folowing tools ?

Reviewer: ```suggestion
> 1. What are the files you are interested in for the following tools ?
```

These 2 pictures are only exemple it is normal if you don't have exactly the same output.

> <question-title>Question</question-title>
>
> 1. How many biodiversity indicators do you have ?
>
> > <solution-title>Solution<solution-title>
> >
> > 1. You should have 7 of them (Shannon, Renyi, Prao, Pielou, Hill, CRE, Berger-Parker). If you have only 5 of them, no problem, it just means you data are too small to compute CRE and Pielou but you can still continue your analysis.
> >
> {: .solution}
>
{: .question}
</div>

<div class="Canopy" markdown="1">
## **Compute a PCA**

> <hands-on-title>Hands-on: Principal components analysis for remote sensing data</hands-on-title>
>
> 1. {% tool [Compute a PCA](toolshed.g2.bx.psu.edu/repos/ecology/srs_pca/srs_pca/0.0.1) %} with the following parameters:
>    - {% icon param-select %} *"In which format are your data ?"*: 'The data you are using are in a zip folder Reflectance'
>      - {% icon param-file %} *"Input data"*: `Reflectance` (output of **Preprocessing sentinel 2 data** {% icon tool %})
>    - {% icon param-select %} *"Do you want to do a PCA or a SPCA ?"*: 'PCA'
>
>

> >    > <tip-title>Tip: If you want to use your own files ENVI BIL<h/tip-title>

Reviewer: ```suggestion
>    > <tip-title>Tip: If you want to use your own files ENVI BIL</tip-title>
```

>      - {% icon param-file %} *"Input data"*: `Reflectance` (output of **Preprocessing sentinel 2 data** {% icon tool %})
>
>    - {% icon param-select %} *"Input the type of indice you want"*: 'NDVI'
>
>    - {% icon param-select %} *"Do you want the raster layer of the indice as an output ?"*: 'No'
>
>
>    > <tip-title>Tip: If you want to use your own files ENVI BIL</tip-title>
>    >
>    > **TODO**: *Check that the "Input raster" datatype is bil and that "Input raster header" datatype is hdr*
>    >
>    > * Go on your raster data 
>    > * Click on {% icon galaxy-pencil %} to edit it
>    > * Click on {% icon galaxy-chart-select-data %} Datatypes
>    > * On "New Type" **Select** bil 
>    > * Press **Save**
>    >
>    >
>    >    > <comment-title>Comment</comment-title>
>    >    >
>    >    > Do the same for the raster header with the datatype hdr
>    >    {: .comment}
>    >
>    {: .tip}
>
>    > <comment-title>Comment</comment-title>
>    >
>    > You can choose whichever indice you want
>    >
>    {: .comment}
>
{: .hands_on}

![ndvi](../../images/remote_sensing/ndvi.png "Normalized different vegetation index tabular")

> > <detail-title>Spectral indices and Essential Biodiversity Variables</detail-title>

Reviewer: ```suggestion
> <details-title>Spectral indices and Essential Biodiversity Variables</details-title>
```

>    >
>    > * Go on your raster data
>    > * Click on {% icon galaxy-pencil %} to edit it
>    > * Click on {% icon galaxy-chart-select-data %} Datatypes
>    > * On "New Type" **Select** bil
>    > * Press **Save**
>    >
>    >
>    >    > <comment-title></comment-title>
>    >    >
>    >    > Do the same for the raster header with the datatype hdr
>    >    {: .comment}
>    >
>    {: .tip}
>
>
>    > <tip-title>You can use the output of **Compute a PCA**</tip-title>
>    >
>    > - {% icon param-select %} *"In which format are your data ?"*: 'Your already have the files in ENVI BIL format'
>    > - {% icon param-file %} *"Input raster"*: `PCA raster` (output of **Compute a PCA** {% icon tool %})
>    > - {% icon param-file %} *"Input header"*: `PCA header` (output of **Compute a PCA** {% icon tool %})
>    > - {% icon param-text %} *"Write a number of the value of alpha"*: '1'
>    >
>    >
>    > ***TODO***: *Here again check that the "Input raster" datatype is bil and that "Input raster header" datatype is hdr*
>    >
>    {: .tip}
>
{: .hands_on}

![Global biodiveristy tabular](../../images/remote_sensing/rasterdivtab.png "Global biodiveristy tabular")

> ![Global biodiveristy graph](../../images/remote_sensing/rasterdiv.png "Global biodiveristy graph")

Reviewer: oh, and could you update the alt texts (the bit between the `[..]` ) for the images to be a bit more descriptive? Imagine you are describing what is in the image to a blind person

Reviewer 2: We have a short guide on [what's appropriate for alt-text vs captions in the GTN Contributing Guide](https://training.galaxyproject.org/training-material/topics/contributing/tutorials/create-new-tutorial-content/tutorial.html#guidelines-on-alt-vs-figcaption-text) which has more details.

---
layout: tutorial_hands_on

title: Sentinel 2 biodiversity
zenodo_link: ''
questions:
- How to get spectral and biodiversity indicators from remote sensing data ?
- Which kind of ecosystem are you studying ? global or canopy data ?
objectives:
- Getting Sentinel 2 data and reformatting them in a generalized way
- Computing spectral indices such as the NDVI
- Calculating and visualizing biodiversity indicators
- Comparing with in-situ data
time_estimation: 48H

> key_points:

Reviewer: Last request: Key points should be conclusions from the tutorial that are important to remember. Think about what the lessons are you want learners to have learned from this tutorial. So can you rephrase them to be full sentences? Things like "It is always important to do X because ..", "Always keep in mind X when interpreting Y", "X is a method to do Y", etc.

---
layout: tutorial_hands_on

title: Sentinel 2 biodiversity
zenodo_link: ''
questions:
- How to get spectral and biodiversity indicators from remote sensing data ?
- Which kind of ecosystem are you studying ? global or canopy data ?
objectives:
- Getting Sentinel 2 data and reformatting them in a generalized way
- Computing spectral indices such as the NDVI
- Calculating and visualizing biodiversity indicators
- Comparing with in-situ data
time_estimation: 48H
key_points:

> - Remote sensing data can be usefull to find hotspot of biodiversity.

Reviewer: ```suggestion
- Remote sensing data can be useful to find hotspots of biodiversity.
```
